### DISCUSSION TITLE
[appDir] searchParams should be available on api handlers as in pages

### DISCUSSION THREAD

**Discussion by volnei** [vercel/next.js#51822] [App Router]
### Goals

1. Make api handlers more cohesive with pages 

This will reduce the api routes boilerplate to handle searchParams making the params avaiable in api route handler.

```js
export const GET = (request: Request, {params, searchParams}) {

}
```

### Background

Current alternative is getting searchparams directly from request url.

### Proposal

I dont know

**Comment by icyJoseph** [comment]
The searchParams are also in the `NextRequest` object.

```ts
import { NextRequest } from "next/server";

export const GET = (request: NextRequest) => {
  const searchParams = request.nextUrl.searchParams;
};

```

**Reply by baba43** [reply]
Hello, searchParams is empty when hosting on vercel.
